libphal/phal_dump.C

```diff
@@ -131,6 +131,30 @@ struct pdbg_target* getProcFromFailingId(const uint32_t failingUnit)
 	return proc;
 }
 
+/**
+ * @brief Function will write reason code and recovery action in a file
+ * @param[in] dumpPath Path of the file
+ * @param[in] reasonCode Reason code of sbe failure
+ * @param[in] recoveryAction Reovery action code for sbe failure
+ *
+ * @return NULL
+ *
+ */
+void writeSbeData(const std::filesystem::path& dumpPath, uint32_t reasonCode,
+                  uint32_t recoveryAction)
+{
+	//Writing data to file
+	std::string dumpFilename = "p10_sbe_data_file";
+	std::filesystem::path basePath = dumpPath / dumpFilename;
```

Comment:
You can put this in a folder names additional_data or similar so it looks cleaner
on the script side instead of checking for the dump type pick whatever in the additional data folder